### ISSUE TITLE
Crash on startup

### ISSUE THREAD

**Issue by knochenhans** [duplicati/duplicati#3068]
- [x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.2.19
- **Operating system**: Arch Linux
- **Backend**: local

## Description

As of today, duplicati crashes when I start it via its systemd user daemon

## Steps to reproduce

1. Starting duplicati via systemctl --user start duplicati 

- **Actual result**:
duplicati crashes after a few seconds
- **Expected result**:
duplicati running normally

## Debug log

```Feb 25 18:58:08 archdesktop mono[8913]:   at Duplicati.Library.Utility.AsyncHttpRequest+AsyncWrapper.GetResponseOrStream () [0x0004d] in <ad02e7cabfcf4629a1895dd6e7a17709>:0
Feb 25 18:58:08 archdesktop mono[8913]:   at Duplicati.Library.Utility.AsyncHttpRequest.GetResponse () [0x00044] in <ad02e7cabfcf4629a1895dd6e7a17709>:0
Feb 25 18:58:08 archdesktop mono[8913]:   at Duplicati.GUI.TrayIcon.HttpServerConnection.PerformRequestInternal[T] (System.String method, System.String endpoint, System.Collections.Generic.Dictionary`2[TKey,TValue] queryparams) [0x001c5] in <a22eddd598d84f91ae3f0950b0581af8>:0
Feb 25 18:58:08 archdesktop mono[8913]:   at Duplicati.GUI.TrayIcon.HttpServerConnection.PerformRequest[T] (System.String method, System.String urlfragment, System.Collections.Generic.Dictionary`2[TKey,TValue] queryparams) [0x00035] in <a22eddd598d84f91ae3f0950b0581af8>:0
Feb 25 18:58:08 archdesktop mono[8913]:   at Duplicati.GUI.TrayIcon.HttpServerConnection.UpdateStatus () [0x00001] in <a22eddd598d84f91ae3f0950b0581af8>:0
Feb 25 18:58:08 archdesktop mono[8913]:   at Duplicati.GUI.TrayIcon.HttpServerConnection..ctor (System.Uri server, System.String password, System.Boolean saltedpassword, System.Boolean dbPasswordSourceDatabase, System.Collections.Generic.Dictionary`2[TKey,TValue] options) [0x000df] in <a22eddd598d84f91ae3f0950b0581af8>:0
Feb 25 18:58:08 archdesktop mono[8913]:   at Duplicati.GUI.TrayIcon.Program.RealMain (System.String[] _args) [0x00454] in <a22eddd598d84f91ae3f0950b0581af8>:0
Feb 25 18:58:08 archdesktop mono[8913]: Request error: System.Net.WebException: Error: ConnectFailure (Network is unreachable) ---> System.Net.Sockets.SocketException: Network is unreachable
Feb 25 18:58:08 archdesktop mono[8913]:   at System.Net.Sockets.Socket.Connect (System.Net.EndPoint remoteEP) [0x000b6] in <640558357cc84390be8186623166329d>:0
Feb 25 18:58:08 archdesktop mono[8913]:   at System.Net.WebConnection.Connect (System.Net.HttpWebRequest request) [0x0016d] in <640558357cc84390be8186623166329d>:0
Feb 25 18:58:08 archdesktop mono[8913]:    --- End of inner exception stack trace ---```

**Comment by warwickmm** [comment]
I'm assuming this has been fixed since I can't reproduce this.  If this still occurs, please reopen and provide more details on how Duplicati was installed (for example, which AUR repository), etc.

**Comment by jwundrak** [comment]
Hi @warwickmm ,

I get the same issue using ArchLinux. I use current version https://aur.archlinux.org/packages/duplicati-latest/ (2.0.5-114-6).  

After long time (3-4 Months), I start duplicati. First time, the start-progress was successful, but the FTP-Source was not correct. After correct the source on remote site (rename share-folder correct), I run into the same issue as @knochenhans .

Running directly the client:

~~~
#> /usr/bin/mono /opt/duplicati-latest/Duplicati.GUI.TrayIcon.exe --webservice-port=8200

Request error: System.Net.WebException: Error: ConnectFailure (Connection refused) ---> System.Net.Sockets.SocketException: Connection refused
  at System.Net.Sockets.SocketAsyncResult.CheckIfThrowDelayedException () [0x00014] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
  at System.Net.Sockets.Socket.EndConnect (System.IAsyncResult asyncResult) [0x0002c] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
  at System.Net.WebConnection+<>c.<Connect>b__16_1 (System.IAsyncResult asyncResult) [0x00006] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
  at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (System.IAsyncResult iar, System.Func`2[T,TResult] endFunction, System.Action`1[T] endAction, System.Threading.Tasks.Task`1[TResult] promise, System.Boolean requiresSynchronization) [0x00019] in <62b430b945fa49a19a75382ef03e7bed>:0 
--- End of stack trace from previous location where exception was thrown ---

  at System.Net.WebConnection.Connect (System.Net.WebOperation operation, System.Threading.CancellationToken cancellationToken) [0x001e1] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
   --- End of inner exception stack trace ---
  at System.Net.WebConnection.Connect (System.Net.WebOperation operation, System.Threading.CancellationToken cancellationToken) [0x0025d] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
  at System.Net.WebConnection.InitConnection (System.Net.WebOperation operation, System.Threading.CancellationToken cancellationToken) [0x000cc] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
  at System.Net.WebOperation.Run () [0x0009a] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
  at System.Net.WebCompletionSource`1[T].WaitForCompletion () [0x00094] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
  at System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.Task`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Func`1[TResult] aborted, System.Threading.CancellationTokenSource cts) [0x000f8] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
  at Duplicati.Library.Utility.AsyncHttpRequest+AsyncWrapper.GetResponseOrStream () [0x0004d] in <1b35de2536e5489e93307d4ad0240fd2>:0 
  at Duplicati.Library.Utility.AsyncHttpRequest.GetResponse () [0x00044] in <1b35de2536e5489e93307d4ad0240fd2>:0 
  at Duplicati.GUI.TrayIcon.HttpServerConnection.PerformRequestInternal[T] (System.String method, System.String endpoint, System.Collections.Generic.Dictionary`2[TKey,TValue] queryparams) [0x001d1] in <3441529c8c09405cbaac2131273162e1>:0 
  at Duplicati.GUI.TrayIcon.HttpServerConnection.PerformRequest[T] (System.String method, System.String urlfragment, System.Collections.Generic.Dictionary`2[TKey,TValue] queryparams) [0x00035] in <3441529c8c09405cbaac2131273162e1>:0 
  at Duplicati.GUI.TrayIcon.HttpServerConnection.UpdateStatus () [0x00001] in <3441529c8c09405cbaac2131273162e1>:0 
  at Duplicati.GUI.TrayIcon.HttpServerConnection..ctor (System.Uri server, System.String password, System.Boolean saltedpassword, Duplicati.GUI.TrayIcon.Program+PasswordSource passwordSource, System.Boolean disableTrayIconLogin, System.Collections.Generic.Dictionary`2[TKey,TValue] options) [0x000cb] in <3441529c8c09405cbaac2131273162e1>:0 
  at Duplicati.GUI.TrayIcon.Program.StartTray (System.String[] _args, System.Collections.Generic.Dictionary`2[TKey,TValue] options, System.String toolkit, Duplicati.GUI.TrayIcon.HostedInstanceKeeper hosted, System.String password, System.Boolean saltedpassword) [0x00024] in <3441529c8c09405cbaac2131273162e1>:0 
Request error: System.Net.WebException: Error: ConnectFailure (Connection refused) ---> System.Net.Sockets.SocketException: Connection refused
  at System.Net.Sockets.SocketAsyncResult.CheckIfThrowDelayedException () [0x00014] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
  at System.Net.Sockets.Socket.EndConnect (System.IAsyncResult asyncResult) [0x0002c] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
  at System.Net.WebConnection+<>c.<Connect>b__16_1 (System.IAsyncResult asyncResult) [0x00006] in <6804f42afa7e4ede9b2d3908165ddc6b>:0 
  at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (System.IAsyncResult iar, System.Func`2[T,TResult] endFunction, System.Action`1[T] endAction, System.Threading.Tasks.Task`1[TResult] promise, System.Boolean requiresSynchronization) [0x00019] in <62b430b945fa49a19a75382ef03e7bed>:0 
--- End of stack trace from previous location where exception was thrown ---


... and so on
~~~

Remote is a QNAP-NAS using FTP. Reverting connection-settings (invalid path) does not help.

What addtional information you need, to resolve this behavior? Starting systemd-Service as root is successful, but the settings are gone.

**Comment by warwickmm** [comment]
Are you saying that the server initially started up fine?  You were able to edit the FTP settings using the web UI, but afterwards the server will no longer start?  If so, did anything else change in between?

What version of mono?

I would also recommend posting in the forums to see if anyone else has encountered this issue:

https://forum.duplicati.com/

**Comment by jwundrak** [comment]
@warwickmm  : Thanks for fast reply.

~~~
#> mono -V
Mono JIT compiler version 6.12.0 (makepkg/c621c35ffa0 Thu Jun 17 02:48:02 PM -03 2021)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          supported, not enabled.
	Suspend:       hybrid
	GC:            sgen (concurrent by default)
~~~

More Detail: I had to reset my NAS (before 2 Months), after that the path was `/backup`. In Duplicati `/backups` was configured. I change the share name to backups and after then, the server was killed, but I'm not sure, if this was the reason.

After start the service as root, no configuration was available. Reset the `~/.config/Duplicati`have no effects.
Have a backup of each start of this directory.

Is there a parameter, to get more details into the stack-trace?

Thanks for forum-link, I'll search and/or create a new thread.

**Comment by warwickmm** [comment]
> After start the service as root, no configuration was available. 

When starting as root, the configurations will be stored elsewhere, possibly `/root/.config/Duplicati`

**Comment by jwundrak** [comment]
`/root/.config/Duplicati` not exists. Starting as root means "default user" instead of `systemctl --user start`use `systemctl start`. This is not root. Sorry for the wrong description.

On Arch, the aur create user *duplicati* with home `/var/lib/duplicati`, but removing whole directory do nothing. Also on restart on "default" user-.

**Comment by warwickmm** [comment]
Does this work or do you get the same error?
```
$ /usr/bin/mono /opt/duplicati-latest/Duplicati.Server.exe --webservice-port=8200
```

**Comment by jwundrak** [comment]
As normal "user" this works. I don't see any errors and the webclient can connect to the server..

**Comment by warwickmm** [comment]
I also use Arch with the duplicati-latest AUR.  I start duplicati with `systemctl --user start duplicati` and don't have any problems with the systemd service files. 

The tray icon does require that the `notify-sharp` package is installed, but I would think that you would encounter a different error message.

I'd recommend creating a post in the forums to see if any other users have encountered this issue.

**Comment by ts678** [comment]
I tested a Manjaro system, which is as close to Arch as I have. I installed Duplicati from AUR using Manjaro's Pamac.

For @knochenhans 

> Starting duplicati via systemctl --user start duplicati

Came up fine, opened the browser, gave me the GUI.

For @jwundrak a similar manual start doesn't show the error, but does complain about root Firefox on user session.

`/usr/bin/mono /opt/duplicati/Duplicati.GUI.TrayIcon.exe --webservice-port=8200`

Manually opening Firefox lets me get Duplicati GUI by browsing to URL `http://localhost:8200` which is its default.
This is an odd configuration. Usually Duplicati.Server.exe runs as root, with user TrayIcon using `--no-hosted-server`.

AUR packagers (not Duplicati project) invented the systemd unit file that runs Duplicati as a special user, using config

```
User=duplicati
Group=duplicati
```

A normal `systemctl start duplicati` comes up, and About --> System info reports that `UserName : duplicati`

So still seems not reproducible, and I couldn't find that a forum post was created (just guessing from the user name).

It's been a few years. Maybe something was fixed, or maybe my test wasn't just right. If nothing more, this will close.
